### ISSUE TITLE
Fix focus style of social icons with no link yet

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#64972]
### Description

Noticed while testing https://github.com/WordPress/gutenberg/pull/64883

When adding social icons, the focus style and the state of a social icon that hasn't a link set yet are problematic.

- The focus style is altered by the opacity set on the social icon. The social icon button is focused but the focus style isn't standard. This issue is very similar to the one fixed in https://github.com/WordPress/gutenberg/pull/62480. No matter what is the state of a control, the focus style needs to be clearly visible and standard.

Screenshot: The Instagram icon isn't linked yet. Notice the opacity alters the focus style. Instead, the GitHub icon is linked. The focus style is the standard one.

![Screenshot 2024-09-02 at 09 38 05](https://github.com/user-attachments/assets/566d7105-4863-4552-bffb-51a376168b67)

- The state of a non-linked social icon is only communicated visually by the opacity. This is insufficient and not accessible as the state is only communicated via a different color produced by the opacity. There's no other indication od the different state of the icons. Color alone is not sufficient. The different state must be communicated via a _shape_.

Screenshot: the GitHub icon does have a link set, the Instagram one doesn't:

![Screenshot 2024-09-02 at 09 33 25](https://github.com/user-attachments/assets/2cd21ad6-dc4e-4ce5-a524-242e50d30263)

@WordPress/gutenberg-design please provide a new design where the state is communicated via a meaningful, understandable sahpe.

### Step-by-step reproduction instructions

- Add a Social icons block.
- Add an icon with a link and another icon without a link.
- Compare the visuals of the different states.
- Compare the focus style of the two icons.

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by afercia** [comment]
How about using the warning icon and removing any opacity? Roughly:

![Screenshot 2024-09-02 at 16 10 49](https://github.com/user-attachments/assets/3ae71124-b8b8-479e-a64c-5f8076681a02)

Maybe a new icon to indicate a missing link would be more meaningful.

**Comment by t-hamano** [comment]
As for the design, I don't have any good ideas at the moment, but apart from that, is there any need to improve the way screen readers read when no links are set? Unlike other blocks, I think the social icon block is expected to always have links applied.

**Comment by afercia** [comment]
>  is there any need to improve the way screen readers read when no links are set?

In the editor: the icon should render even with no link set and an appropriate visual indication and label should clearly communicate the icon misses a link.
On the front end: I'd tend to think a social icon with no link should no render at all.

**Comment by hanneslsm** [comment]
> How about using the warning icon and removing any opacity? Roughly:

Interesting idea. The icons block is definitely a special case, but let's not forget we have also blocks that currently use a opacity when they are not rendered in the frontend, like the "Read more" link in a Post Template or "Previous Page" and "Next Page" in the Pagination block.
<img width="661" alt="CleanShot 2024-09-04 at 09 25 44@2x" src="https://github.com/user-attachments/assets/1bc2cc07-6e4f-4263-96b7-482e7013ef34">
And then, of course, there are image placeholders without opacity.

I guess what I'm trying to say is that we should look holistically from UX and UI standpoint at that topic "placeholders" / what's displayed on the editor but not frontend.

**Comment by karmatosed** [comment]
I think this needs a unified approach over just jumping into solutions. For now I am going to remove the needs design label, what I would recommend is looking for states that indicate in other cases 'needing link' and we can perhaps unify the language a bit across all.
